File: src/AddressBook/ListPeople.cs

using System;
using System.IO;

namespace Google.ProtocolBuffers.Examples.AddressBook {
  class ListPeople {
    /// <summary>
    /// Iterates though all people in the AddressBook and prints info about them.
    /// </summary>
    static void Print(AddressBook addressBook) {
      foreach (Person person in addressBook.PersonList) {
        Console.WriteLine("Person ID: {0}", person.Id);
        Console.WriteLine("  Name: {0}", person.Name);
        if (person.HasEmail) {
          Console.WriteLine("  E-mail address: {0}", person.Email);
        }

        foreach (Person.Types.PhoneNumber phoneNumber in person.PhoneList) {
          switch (phoneNumber.Type) {
            case Person.Types.PhoneType.MOBILE:
              Console.Write("  Mobile phone #: ");
              break;
            case Person.Types.PhoneType.HOME:
              Console.Write("  Home phone #: ");
              break;
            case Person.Types.PhoneType.WORK:
              Console.Write("  Work phone #: ");
              break;
          }
          Console.WriteLine(phoneNumber.Number);
        }
      }
    }

    /// <summary>
    /// Entry point - loads the addressbook and then displays it.
    /// </summary>
    public static int Main(string[] args) {
      if (args.Length != 1) {
        Console.Error.WriteLine("Usage:  ListPeople ADDRESS_BOOK_FILE");
        return 1;
      }

      if (!File.Exists(args[0])) {
        Console.WriteLine("{0} doesn't exist. Add a person to create the file first.", args[0]);
        return 0;
      }

      // Read the existing address book.
      using (Stream stream = File.OpenRead(args[0])) {
        AddressBook addressBook = AddressBook.ParseFrom(stream);
        Print(addressBook);
      }
      return 0;
    }
  }
}

File: kotlin/protobuf/example/src/main/kotlin/com/example/tutorial/ListPeople.kt
/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.tutorial

import com.example.tutorial.AddressBookProtos.AddressBook
import java.io.FileInputStream
import kotlin.system.exitProcess

object ListPeople {
  fun print(addressBook: AddressBook) {
    for (person in addressBook.personList) {
      println("Person ID: ${person.id}")
      println("  Name: ${person.name}")
      if (person.hasEmail()) {
        println("  E-mail address: ${person.email}")
      }
      for (phoneNumber in person.phoneList) {
        when (phoneNumber.type) {
          AddressBookProtos.Person.PhoneType.MOBILE -> print("  Mobile phone #: ")
          AddressBookProtos.Person.PhoneType.HOME -> print("  Home phone #: ")
          AddressBookProtos.Person.PhoneType.WORK -> print("  Work phone #: ")
        }
        println(phoneNumber.number)
      }
    }
  }

  @JvmStatic
  fun main(args: Array<String>) {
    if (args.size != 1) {
      System.err.println("Usage:  ListPeople ADDRESS_BOOK_FILE")
      exitProcess(-1)
    }
    print(FileInputStream(args.single()).use {
      AddressBook.parseFrom(it)
    })
  }
}